File: lib/data/models/vinculo_cuidador.dart
/// Vinculo somente-leitura entre um cuidador e um idoso/tutelado.
class VinculoCuidador {
  final String idosoId;
  final String idosoNome;

  VinculoCuidador({required this.idosoId, required this.idosoNome});

  factory VinculoCuidador.fromJson(Map<String, dynamic> json) => VinculoCuidador(
        idosoId: json['idosoId'] ?? '',
        idosoNome: json['idosoNome'] ?? '',
      );
}

/// Resumo (somente leitura) do progresso de um idoso vinculado.
class ResumoIdoso {
  final String idosoNome;
  final int cursosConcluidos;
  final int sequenciaDias;
  final int pontos;

  ResumoIdoso({
    required this.idosoNome,
    required this.cursosConcluidos,
    required this.sequenciaDias,
    required this.pontos,
  });

  factory ResumoIdoso.fromJson(Map<String, dynamic> json) => ResumoIdoso(
        idosoNome: json['idosoNome'] ?? '',
        cursosConcluidos: json['cursosConcluidos'] ?? 0,
        sequenciaDias: json['sequenciaDias'] ?? 0,
        pontos: json['pontos'] ?? 0,
      );
}

File: backend-java/src/main/java/com/smarthas/backend/model/VinculoCuidador.java
package com.smarthas.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

/**
 * Vinculo somente-leitura entre um cuidador e um idoso/tutelado: o cuidador
 * acompanha o progresso, nunca age em nome da pessoa (a conta do idoso
 * continua sendo dele, com seu proprio login).
 */
@Entity
@Table(name = "vinculos_cuidador")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class VinculoCuidador {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long cuidadorId;

    @Column(nullable = false)
    private Long idosoId;

    @Column(nullable = false)
    @Builder.Default
    private String status = "ATIVO";

    @Column(nullable = false)
    @Builder.Default
    private LocalDateTime criadoEm = LocalDateTime.now();
}
